<a href="https://colab.research.google.com/github/wenjunsun/Covid-19-analysis-with-uw-ubicomp/blob/master/analysis/compare_30_days_before_and_after.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# What we want to do?
In this notebook we want to investigate that how people's behavior in terms of home dwell time, distance traveled, etc changed after the first case of covid happend in their county and after shelter in place was issued in the county/state.

Specifically we will compare average behavior data of 30 days before and 30 days after first case/shelter in place.  

# load data and split it into data that contains 30 days before and 30 days after first case/shelter in place


In [ ]:
cd drive/My\ Drive/COVID\ 19\ data\ analysis

In [ ]:
ls

 aggreagate_data_on_county.ipynb
 agg_social_dist
 analysis_and_graphs.ipynb
 compare_30_days_before_and_after.ipynb
'Copy of COVID-19 US state policy database.csv'
 days_since.csv
 merge_mobility_with_first_case_and_shelter_in_place.ipynb
 small_data.csv
 social_dist_all_trimmed.csv
'Social Distancing Index Exploration.ipynb'
 social_dist_with_dates
 social_dist_with_days_since


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
data = pd.read_csv("agg_social_dist")

In [ ]:
data.head()

Unnamed: 0  ... distance_traveled_from_home
0           0  ...                22780.000000
1           1  ...                89197.250000
2           2  ...               140597.925311
3           3  ...                60138.741750
4           4  ...                 8579.466002

[5 rows x 19 columns]

In [ ]:
data.describe()

Unnamed: 0  ...  distance_traveled_from_home
count  332517.000000  ...                 3.317900e+05
mean   166258.000000  ...                 1.285544e+04
std     95989.534068  ...                 1.890418e+04
min         0.000000  ...                 1.250000e+02
25%     83129.000000  ...                 7.761239e+03
50%    166258.000000  ...                 1.007196e+04
75%    249387.000000  ...                 1.381926e+04
max    332516.000000  ...                 1.211508e+06

[8 rows x 15 columns]



*   We can see that the mean of days_since_first_case is -22. This means that the majority of our data are about before the first case appears in a county
*   We can see the mean of device_count = 6395, so on average we measured behaviors of about 6395 people per county. Interesting. However the standard deviation is huge, I guess some counties we have a lot of data on (every person has a phone) and other county less so. The max is 412936 devices, that must be a huge county
*   mean home_dwell_time is 647 minutes = 10 hours. interesting. mean percentage_time_home = 86% 



In [ ]:
# want to see which county we have device_count of 412936
data[data['device_count'] == 412936] 

Unnamed: 0  ... distance_traveled_from_home
164926      164926  ...                11075.470489

[1 rows x 19 columns]

So we have a lot of devices in LA county which kind of make sense. It is a big county with a lot of people

In [ ]:
data_of_30_days_before_first_case = data[(data['days_since_first_case'] < 0) & (data['days_since_first_case'] >= -30)]

In [ ]:
data_of_30_days_after_first_case = data[(data['days_since_first_case'] >= 0) & (data['days_since_first_case'] <= 30)]

In [ ]:
data_of_30_days_before_shelter = data[(data['days_since_shelter'] < 0) & (data['days_since_shelter'] >= -30)]

In [ ]:
data_of_30_days_after_shelter = data[(data['days_since_shelter'] >= 0) & (data['days_since_shelter'] <= 30)]

# Compare data of 30 days before and 30 days after first case

In [ ]:
data_of_30_days_before_first_case.head()

Unnamed: 0  ... distance_traveled_from_home
1325        1325  ...                10352.586596
2269        2269  ...                10346.956188
2464        2464  ...                12042.490755
2474        2474  ...                 8743.759775
2487        2487  ...                16360.121921

[5 rows x 19 columns]

In [ ]:
data_of_30_days_after_first_case.tail()

Unnamed: 0  ... distance_traveled_from_home
332499      332499  ...                 1181.697761
332504      332504  ...                26941.522277
332506      332506  ...               446930.986726
332509      332509  ...                 3705.877451
332510      332510  ...                 4472.597315

[5 rows x 19 columns]

In [ ]:
print(f"number of rows of 30 days before first case: {data_of_30_days_before_first_case.shape[0]}")
print(f"number of rows of 30 days after first case: {data_of_30_days_after_first_case.shape[0]}")

number of rows of 30 days before first case: 78187
number of rows of 30 days after first case: 76605


so we pretty much have the same amounts of data points in the 30 days before period and the 30 days after period, so won't suffer imbalanced data points bias

In [ ]:
data_of_30_days_before_first_case.isnull().sum()

Unnamed: 0                             0
date_range_start                       0
date_range_end                         0
state                                  0
state_code                             0
cnamelong                              0
county_code                            0
days_since_first_case                  0
days_since_shelter                     0
candidate_device_count                 0
device_count                           0
completely_home_device_count           0
part_time_work_behavior_devices        0
full_time_work_behavior_devices        0
delivery_behavior_devices              0
median_home_dwell_time                 0
median_non_home_dwell_time         77298
median_percentage_time_home        77298
distance_traveled_from_home           19
dtype: int64

In [ ]:
data_of_30_days_after_first_case.isnull().sum()

Unnamed: 0                             0
date_range_start                       0
date_range_end                         0
state                                  0
state_code                             0
cnamelong                              0
county_code                            0
days_since_first_case                  0
days_since_shelter                     0
candidate_device_count                 0
device_count                           0
completely_home_device_count           0
part_time_work_behavior_devices        0
full_time_work_behavior_devices        0
delivery_behavior_devices              0
median_home_dwell_time                 0
median_non_home_dwell_time         69079
median_percentage_time_home        69079
distance_traveled_from_home          279
dtype: int64

we can see from the above data frame that we only have 5-10% of recorded median_non_home_dwell_time and median_percentage_time_home (more than 90% null data), but we have perfect record of median_home_dwell_time, so we can just pick median_home_dwell_time to analyze

In [ ]:
# statistics about 30 days before first case
data_of_30_days_before_first_case.describe()[['candidate_device_count',\
                                              'device_count', 'completely_home_device_count',\
                                              'part_time_work_behavior_devices',\
                                              'full_time_work_behavior_devices',\
                                              'delivery_behavior_devices',\
                                              'median_home_dwell_time',\
                                              'distance_traveled_from_home']]

candidate_device_count  ...  distance_traveled_from_home
count            78187.000000  ...                 7.816800e+04
mean                14.195199  ...                 1.339026e+04
std                171.622187  ...                 2.405520e+04
min                  0.000000  ...                 2.320000e+02
25%                  0.000000  ...                 8.059154e+03
50%                  0.000000  ...                 1.035197e+04
75%                  0.000000  ...                 1.400087e+04
max               5627.000000  ...                 1.211508e+06

[8 rows x 8 columns]

In [ ]:
# statistics about 30 days after first case
data_of_30_days_after_first_case.describe()[['candidate_device_count',\
                                              'device_count', 'completely_home_device_count',\
                                              'part_time_work_behavior_devices',\
                                              'full_time_work_behavior_devices',\
                                              'delivery_behavior_devices',\
                                              'median_home_dwell_time',\
                                              'distance_traveled_from_home']]

candidate_device_count  ...  distance_traveled_from_home
count            76605.000000  ...                 7.632600e+04
mean               257.999269  ...                 1.147974e+04
std               1093.952137  ...                 2.111861e+04
min                  0.000000  ...                 1.250000e+02
25%                  0.000000  ...                 6.817062e+03
50%                  0.000000  ...                 8.746437e+03
75%                  0.000000  ...                 1.165161e+04
max              30503.000000  ...                 1.204601e+06

[8 rows x 8 columns]

from these 2 crude statistics table we could aldready spot some differences between people's behaviors 30 days before and 30 days after the first case:
1. don't know what candiate device means but it increases dramatically after first case
2. mean completely home device count increases from 1525 to 2351  

Let's analyze \\
1. completely_home_device_percentage
2. part_time_work_behavior_percentage
3. full_time_work_behavior_percentage
4. delivery_behavior_percentage
5. median_home_dwell_time
6. mean_distance_traveled_from_home
30 days before and 30 days after first case.
where percentage just means divide the column by device count.

In [ ]:
# 1, 2, 3, 4analysis
columns = ['completely_home_device_count', 'part_time_work_behavior_devices', \
           'full_time_work_behavior_devices', 'delivery_behavior_devices']
for col in columns:
  before_behavior_percentage = data_of_30_days_before_first_case[col].sum() / \
    data_of_30_days_before_first_case['device_count'].sum()
  after_behavior_percentage = data_of_30_days_after_first_case[col].sum() / \
    data_of_30_days_after_first_case['device_count'].sum()
  print(f"30 days before first case percentage of {col}: {before_behavior_percentage}")
  print(f"30 days after first case percentage of {col}: {after_behavior_percentage}")
  print()

30 days before first case percentage of completely_home_device_count: 0.22551973795931396
30 days after first case percentage of completely_home_device_count: 0.3590801341884025

30 days before first case percentage of part_time_work_behavior_devices: 0.10718879165331115
30 days after first case percentage of part_time_work_behavior_devices: 0.06790782618195314

30 days before first case percentage of full_time_work_behavior_devices: 0.17398280753792122
30 days after first case percentage of full_time_work_behavior_devices: 0.12709269019853267

30 days before first case percentage of delivery_behavior_devices: 2.672284692856611e-05
30 days after first case percentage of delivery_behavior_devices: 0.0004734462854093858



In [ ]:
0.0004734462854093858 / 2.672284692856611e-05

17.71691042780635

1. we can see that people stay at home more after first case appears: home device percentage jumps from 22% to 36%
2. part time work behavior percentage drops from 10% to 6.7%
3. full time behavior drops from 17% to 12.7%
4. delivery behavior device percentage almost increases 18 times \\
which all make sense

In [ ]:
# 5,6 analysis, need to weighted average again.
# since distance and home dwell time are all medians
before_mean_home_dwell_time = (data_of_30_days_before_first_case['median_home_dwell_time'] \
  * data_of_30_days_before_first_case['device_count']).sum() / \
  data_of_30_days_before_first_case['device_count'].sum()
after_mean_home_dwell_time = (data_of_30_days_after_first_case['median_home_dwell_time'] \
  * data_of_30_days_after_first_case['device_count']).sum() / \
  data_of_30_days_after_first_case['device_count'].sum()
print(f"before_mean_home_dwell_time: {before_mean_home_dwell_time}")
print(f"after_mean_home_dwell_time: {after_mean_home_dwell_time}")

before_mean_home_dwell_time: 587.458941304336
after_mean_home_dwell_time: 764.7527937164997


In [ ]:
before_mean_distance = (data_of_30_days_before_first_case['distance_traveled_from_home'] \
  * data_of_30_days_before_first_case['device_count']).sum() / \
  data_of_30_days_before_first_case['device_count'].sum()
after_mean_distance = (data_of_30_days_after_first_case['distance_traveled_from_home'] \
  * data_of_30_days_after_first_case['device_count']).sum() / \
  data_of_30_days_after_first_case['device_count'].sum()
print(f"before_mean_distance: {before_mean_distance}")
print(f"after_mean_distance: {after_mean_distance}")

before_mean_distance: 12689.17195000683
after_mean_distance: 11708.676082523452


In [ ]:
after_mean_home_dwell_time / before_mean_home_dwell_time

1.3017978618531498

In [ ]:
after_mean_distance / before_mean_distance

0.9227297201624847

1. we can see that people stay at home more after first case of COVID-19. increases from 587 minutes per day to 764 minutes per day, an increase of 30%
2. distance traveled by people decreased on average. -> decreased about 8%, although not much. (don't know what unit the distance is in? it will by crazy that everyone travels 12689 miles per day)

# Compare data of 30 days before and 30 days after shelter in place

Now let's repeat the above analyis for 30 days before and 30 days after shelter in place was issued.

In [ ]:
# 1, 2, 3, 4, 6 analysis
columns = ['completely_home_device_count', 'part_time_work_behavior_devices', \
           'full_time_work_behavior_devices', 'delivery_behavior_devices']
for col in columns:
  before_behavior_percentage = data_of_30_days_before_shelter[col].sum() / \
    data_of_30_days_before_shelter['device_count'].sum()
  after_behavior_percentage = data_of_30_days_after_shelter[col].sum() / \
    data_of_30_days_after_shelter['device_count'].sum()
  print(f"30 days before shelter in place, percentage of {col}: {before_behavior_percentage}")
  print(f"30 days after shelter in place, percentage of {col}: {after_behavior_percentage}")
  print()

30 days before shelter in place, percentage of completely_home_device_count: 0.2663601438680131
30 days after shelter in place, percentage of completely_home_device_count: 0.40836784123788233

30 days before shelter in place, percentage of part_time_work_behavior_devices: 0.09294609942856893
30 days after shelter in place, percentage of part_time_work_behavior_devices: 0.05503536756401233

30 days before shelter in place, percentage of full_time_work_behavior_devices: 0.15982911583196868
30 days after shelter in place, percentage of full_time_work_behavior_devices: 0.10221952555880594

30 days before shelter in place, percentage of delivery_behavior_devices: 0.0
30 days after shelter in place, percentage of delivery_behavior_devices: 0.003747411424778401



In [ ]:
# mean_median_home_dwell_time analysis, need to weighted average again
before_mean_home_dwell_time = (data_of_30_days_before_shelter['median_home_dwell_time'] \
  * data_of_30_days_before_shelter['device_count']).sum() / \
  data_of_30_days_before_shelter['device_count'].sum()
after_mean_home_dwell_time = (data_of_30_days_after_shelter['median_home_dwell_time'] \
  * data_of_30_days_after_shelter['device_count']).sum() / \
  data_of_30_days_after_shelter['device_count'].sum()
print(f"before_mean_home_dwell_time: {before_mean_home_dwell_time}")
print(f"after_mean_home_dwell_time: {after_mean_home_dwell_time}")

before_mean_home_dwell_time: 624.3097946995292
after_mean_home_dwell_time: 849.5859743725885


In [ ]:
before_mean_distance = (data_of_30_days_before_shelter['distance_traveled_from_home'] \
  * data_of_30_days_before_shelter['device_count']).sum() / \
  data_of_30_days_before_shelter['device_count'].sum()
after_mean_distance = (data_of_30_days_after_shelter['distance_traveled_from_home'] \
  * data_of_30_days_after_shelter['device_count']).sum() / \
  data_of_30_days_after_shelter['device_count'].sum()
print(f"before_mean_distance: {before_mean_distance}")
print(f"after_mean_distance: {after_mean_distance}")

before_mean_distance: 14890.036775633482
after_mean_distance: 9874.372134492716


the comparison pretty much echos the before and after first case analysis, a little bit differnet but the quanlitative message is the same.\\
Noteworthy things are: comparing to first case comparison, in shelter in place comparisons mean distance traveled by people dramtically decreased from about 15000 to 10000. **That is interesting, we didn't see this in the first case comparison. This means people's traveling behavior changes dramatically after government's stay-at-home policy was issued, not after they have seen the first death of COVID within the county**

# Conclusion:
Comparing average behavior of 30 days before and after shelter in place in counties all over the country we found that
1. people travel siginificantly less after government's stay-at-home policy was issued, but not after first case appeared in their county.
2. home dwell time changed from 624 minutes per day to an average of 849 minutes per day after shelter in place.
3. full time behavior drops from 15% to 10% after shelter in place.
4. % of stay at home devices increase from 26% to 40% after shelter in place.